magprime/__init__.py

```diff
@@ -47,6 +47,11 @@ def set_empty_donation(self):
         if not self.extra_donation:
             self.extra_donation = 0
 
```

Comment:
> What I'd really like is a validation that applies to any cost property that will catch this, so I don't have to write a new one every time I add one of these. ;)

We actually have a ``discount`` cost property which is allowed/intended to be negative, so that wouldn't work.

In theory, we could have ``@non_negative_cost_property`` as a separate decorator or something.  Or maybe ``@cost_property(allow_negatives=True)``?

> Can you specify what you mean when you say instance method?

Short explanation:

```python
def regular_function(): pass

class SomeClass:
    def instance_method(self): pass

    @classmethod
    def class_method(cls): pass

    @staticmethod
    def static_method(): pass
```

I'm happy to explain the difference between the above four types of functions if you'd like.